ewfs/noise_models.py

```diff
@@ -45,3 +55,49 @@ def bitflip_model(p: float) -> NoiseModel:
 
     return noise_bit_flip
 
+
+def get_single_qubit_gates(circ: QuantumCircuit) -> Set[str]:
+    """Extracts single-qubit gates from a quantum circuit.
+
+    Args:
+        circ: Quantum circuit.
+    Returns:
+        List of string names of single-qubit gates.
+    """
+    gate_set = set()
+    for instruction in circ.data:
+        op = instruction.operation
+        if op.num_qubits == 1:
+            if op.name != 'measure':
+                gate_set.add(op.name)
+    return gate_set
+
+
+def get_two_qubit_gates(circ: QuantumCircuit) -> Set[str]:
+    """Extracts two-qubit gates from a quantum circuit.
+
+    Args:
+        circ: Quantum circuit.
+    Returns:
+        List of string names of two-qubit gates.
```

Comment:
[nit]: Returns a set instead of a list

```suggestion
        Set of string names of two-qubit gates.
```